middleware/locale/method/cookie.go

```diff
@@ -0,0 +1,17 @@
+package method
+
+import (
+	"net/http"
+	"strings"
+)
+
+func detectByCookie(r *http.Request, settings *Settings) []string {
+	name := strings.TrimSpace(settings.CookieName)
+	if name == "" {
+		name = "locale"
+	}
```

Comment:
This is best done in the *setup* phase. The whole function would collapse to one line, and be moved into its caller.